src/api.ts

```diff
@@ -474,10 +488,159 @@ export async function proxyRequestToHash( req: any, res: any ) {
 		return;
 	}
 
+	touchCommit( commitHash, runEnv );
```

Comment:
Touching the commit (i.e. marking it as accessed) has been moved to inside the proxy function, when we are really sure we are redirecting requests to the proxy.

Incidentally I fixed the bug were we didn't account for the fact that a commit can have images running in different environments.